lib/ASTGen/Sources/ASTGen/ASTGen.swift

```diff
@@ -0,0 +1,78 @@
+import SwiftParser
+import SwiftSyntax
+
+import CASTBridging
+
+extension Array {
+  public func withBridgedArrayRef<T>(_ c: (BridgedArrayRef) -> T) -> T {
+    withUnsafeBytes { buf in
+      c(BridgedArrayRef(data: buf.baseAddress!, numElements: count))
+    }
+  }
+}
+
+extension UnsafePointer {
+  public var raw: UnsafeMutableRawPointer {
+    UnsafeMutableRawPointer(mutating: self)
+  }
+}
+
+// TODO: :(
+var declContext: UnsafeMutableRawPointer! = nil
```

Comment:
mutable module-wide global variable buried between the lines.

It feels like it supposes to be a `ASTGenVisitor` property set in the `ASTGenVisitor` initializer. Unfortunate `ASTGenVisitor` is a struct that makes it all `mutating`. How about wrapping it in a class

```swift
class DeclContext {
    var declContext: UnsafeMutableRawPointer! = nil
}

struct ASTGenVisitor ... {
    var declContext = DeclContext()
}

```

and yes, nobody asked me for this comment

Reply:
Gaaa, I hadn't noticed that. I've done a Sneaky Property Wrapper Trick for the moment.